src/xmipp/libraries/reconstruction/resolution_fsc.cpp

```diff
@@ -24,202 +24,188 @@
+	addSeeAlsoLine("resolution_ssnr");
+
+	addParamsLine("   -i <input_file>           : either an image/volume or a selection file");
+	addParamsLine("   requires --ref;");
+	addParamsLine("or --set_of_images <selfile> : selfile containing a set of 2D-images");
+	addParamsLine("   [--oroot <root_file=\"\">] : Root of the output metadata. If not set, input file rootname is taken.");
+	addParamsLine("   [-o <output_file=\"\">]   : Output file name.");
+
+	addParamsLine("   [--ref <input_file>]      : filename for reference image/volume");
+	addParamsLine("   [--sampling_rate <Ts=1>]  : Pixel size (Angstrom)");
+	addParamsLine("  alias -s;");
+	addParamsLine("   [--dont_apply_geo]        : for 2D-images: do not apply transformation stored in the header");
+	addParamsLine("   [--do_dpr]                : compute dpr, by default only frc is computed");
+	addParamsLine("   [--max_sam <max_sr=-1>]   : set fsc to 0 for frequencies above this one (Angstrom), -1 -> all fequencies");
+	addParamsLine("                             : --max_sam = 10A -> frequencies higher than 0.1 A^-1 =0");
+	//for R-factor
+	addParamsLine("   [--do_rfactor]            : compute R-factor for input volumes");
+	addParamsLine("   [--min_sam <min_sr=-1>]   : minimum frequency may use for calculating R-factor (Angstrom)");
+	addParamsLine("                             : --min_sam = 10A -> frequencies smaller than 0.1 A^-1 =0");
+
+	addExampleLine("Resolution of subset2.vol volume with respect to subset1.vol reference volume using 5.6 pixel size (in Angstrom):", false);
+	addExampleLine("xmipp_resolution_fsc --ref subset1.vol  -i subset2.vol --sampling_rate 5.6 ");
+	addExampleLine("Resolution of a set of images using 5.6 pixel size (in Angstrom):", false);
+	addExampleLine("xmipp_resolution_fsc --set_of_images selfile.sel --sampling_rate 5.6");
+}
+
+void ProgResolutionFsc::readParams()
+{
+	sam = getDoubleParam("--sampling_rate");
+
+	apply_geo = !checkParam("--dont_apply_geo");
+
+	max_sam = getDoubleParam("--max_sam");
+	do_dpr = checkParam("--do_dpr");
+	do_set_of_images = checkParam("--set_of_images");
+	//for calculating r-factor
+	min_sam = getDoubleParam("--min_sam");
+	do_rfactor = checkParam("--do_rfactor");
+
+	if(do_set_of_images)
+	{
+		fn_sel = getParam("--set_of_images");
+		if (checkParam("-i") || checkParam("--ref"))
+			REPORT_ERROR(ERR_ARG_INCORRECT, "--set_of_images should not be provided with -i or --ref");
+	}
+	else
+	{
+		fn_ref = getParam("--ref");
+		fn_img = getParam("-i");
+	}
+
+	do_o = checkParam("-o");
+	if (do_o)
+		fn_out = getParam("-o");
+	else
+		fn_root = getParam("--oroot");
+}
+
+void ProgResolutionFsc::writeFiles(const FileName &fnRoot,
+				const MultidimArray<double> &freq,
+				const MultidimArray<double> &frc,
+				const MultidimArray<double> &frc_noise,
+				const MultidimArray<double> &dpr,
+				const MultidimArray<double> &error_l2,
+				double max_sam, bool do_dpr, double rFactor)
```

Comment:
at least `max_sam` and `do_dpr` are the class instance attributes. No need to pass them (notice that e.g. `min_sam` is not passed either)